xds/internal/xdsclient/client_refcounted.go

```diff
@@ -55,34 +55,37 @@ func clientRefCountedClose(name string) {
 	}
 	client.clientImpl.close()
 	xdsClientImplCloseHook(name)
-	delete(clients, name)
+	delete(pool.clients, name)
 
 }
 
 // newRefCounted creates a new reference counted xDS client implementation for
 // name, if one does not exist already. If an xDS client for the given name
 // exists, it gets a reference to it and returns it.
-func newRefCounted(name string, config *bootstrap.Config, watchExpiryTimeout, idleChannelExpiryTimeout time.Duration, streamBackoff func(int) time.Duration) (XDSClient, func(), error) {
-	clientsMu.Lock()
-	defer clientsMu.Unlock()
+func newRefCounted(name string, pool *Pool, config *bootstrap.Config, watchExpiryTimeout, idleChannelExpiryTimeout time.Duration, streamBackoff func(int) time.Duration) (XDSClient, func(), error) {
```

Comment:
Same here. Ideally, everything in this file should move to `pool.go` in one form or the other.

Reply:
Done